cmd/kubeadm/app/util/users/users_linux.go

```diff
@@ -0,0 +1,631 @@
+// +build linux
+
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package users
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"sort"
+	"strconv"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"k8s.io/klog/v2"
+	"k8s.io/kubernetes/cmd/kubeadm/app/constants"
+)
+
+// EntryMap holds a map of user or group entries.
+type EntryMap struct {
+	entries map[string]*entry
```

Comment:
do we need anything more than the UID / GID?

if not this can be used:
```
ugs, err := users.AddUsersAndGroups()

uid := ugs.Users[kubeadmconstants.KubeAPIServerUserName].ID()
gid := ugs.Groups[kubeadmconstants.KubeAPIServerUserName].ID()
```

Reply:
https://github.com/kubernetes/kubernetes/pull/102463#discussion_r645119455

Reply:
`uid` and `gid` can end up as e.g. `101` and `102`.

Reply:
I see my understanding of what is stored in Users and Groups in the UsersAndGroups struct was flawed, you comment has clarified that for me. Thanks!

Reply:
actually the syntax is:
```
ugs, err := users.AddUsersAndGroups()

const name := kubeadmconstants.KubeAPIServerUserName
uid := ugs.Users.ID(name)
gid := ugs.Groups.ID(name)
```